frontend/src/metabase/containers/ItemPicker/ItemPicker.tsx

```diff
@@ -150,19 +147,17 @@ function ItemPicker<TId>({
 
     if (searchString) {
       query.q = searchString;
-      if (showOnlyPersonalCollections) {
-        query.filter_items_in_personal_collection = "only";
+      if (filterPersonalCollections) {
+        query.filter_items_in_personal_collection = filterPersonalCollections;
       }
     } else {
       query.collection = openCollectionId;
     }
 
-    if (models.length === 1) {
-      query.models = models;
-    }
+    query.models = models;
```

Comment:
This is caught during unit tests 😉 . It was kind of a bug that we fetche all results instead of just the specified models. There is one case where we set `models = ["dataset", "card"]` and we didn't apply `models` param at all.
#### After the fix
![Screenshot 2023-11-09 at 7 36 58 PM](https://github.com/metabase/metabase/assets/1937582/c4aefe03-b641-4e8c-8745-825494089eb5)

#### Before the fix
![Screenshot 2023-11-09 at 7 37 01 PM](https://github.com/metabase/metabase/assets/1937582/1f717b7e-9c3f-40fd-a30b-cf31fe79dec5)


Although we used to fetch results from all models it doesn't make any difference in `ItemPicker` since FE will filter those results based on `models` anyway.

https://github.com/metabase/metabase/blob/68788d775f5642046630ad34c98c4c5e0a5e9522/frontend/src/metabase/containers/ItemPicker/ItemPickerView.tsx#L185

tl;dr: This change is needed for unit tests and as a result it fetches only necessary data from the API.